Add math formulation docs (#25)
CliMA/Thermodynamics.jl: docs/src/Formulation.md

# Mathematical formulation

The thermodynamics of moist air is often subject to empirical approximations, which usually are opaque, internally inconsistent, and/or inconsistent across model components. For example, microphysical process models often use different approximations for thermodynamic quantities such as saturation vapor pressures than the dynamical core. The often bewildering array of approximations makes it difficult to achieve global conservation, e.g., of energy, and it complicates the use of models for other planetary atmospheres, with different thermodynamic parameters.

> This chapter introduces one consistent set of thermodynamic approximations for all model components. The key to thermodynamic consistency at reasonable accuracy is to take the specific heat capacities of the constituents of moist air (dry air, water vapor, liquid water, and ice) to be constant, i.e., to assume the gases to be calorically perfect. We discuss how to derive all other thermodynamic quantities that are needed on the basis of this one approximation ([Romps2008](@cite),[Marquet2015](@cite)). This includes:

Reviewer: ```suggestion
Here we  introduce one consistent set of thermodynamic approximations for all model components. The key to thermodynamic consistency at reasonable accuracy is to take the specific heat capacities of the constituents of moist air (dry air, water vapor, liquid water, and ice) to be constant, i.e., to assume the gases to be calorically perfect. We discuss how to derive all other thermodynamic quantities that are needed on the basis of this one approximation ([Romps2008](@cite),[Marquet2015](@cite)). This includes:
```

```math
\begin{equation}
\begin{split}\label{e:enthalpy_definition}
    h(T, q)  &= (1-q_t) h_d + q_v h_v + q_l h_l + q_i h_i \\
        &= c_{pm}(q) (T-T_0) + q_v L_{v,0} - q_i L_{f,0} + (1-q_t) R_d T_0\\
        &= I(q, T) + R_m T,
\end{split}
\end{equation}
```
where the last equality used ``c_{pm} = c_{vm} + R_m`` (Eq. \ref{e:SpecificHeatRelation}). The enthalpy is the relevant thermodynamic energy quantity in fluid transport. It arises in boundary conditions for energy fluxes and in the modeling of subgrid-scale (SGS) turbulent transport. For those purposes, we need gradients of the enthalpy, which can be written as
```math
\begin{equation}
\label{e:EnthalpyGradient}
    ∇h = c_{pm}(q) ∇T - h_d(T) ∇q_t
    + h_v(T) ∇q_v + h_l(T) ∇q_l + h_i(T) ∇q_i.
\end{equation}
```
This cleanly separates gradients involving temperature and gradients involving specific humidities.

## Moist Static Energy

The sum of the specific enthalpy of moist air and the specific gravitational potential energy ``Φ`` is the moist static energy [Neelin1987](@cite)
```math
\begin{equation}\label{e:MSE}
\mathrm{MSE} = h + Φ.
\end{equation}
```

> Moist static energy arises naturally as the static energy component that is transported in moist air. ``Static`` here refers to the fact that the (small) kinetic energy contribution to the total energy is neglected (see section TODO). The global integral of moist static energy is approximately conserved in adiabatic processes, even in the presence of reversible phase transitions and latent heat release. It is also approximately materially conserved [Romps2015](@cite).

Reviewer: ```suggestion
Moist static energy arises naturally as the static energy component that is transported in moist air. ``Static`` here refers to the fact that the (small) kinetic energy contribution to the total energy is neglected (see section TODO). The global integral of moist static energy is approximately conserved in adiabatic processes, even in the presence of reversible phase transitions and latent heat release. It is also approximately materially conserved ([Romps2015](@cite)).
```

```

Here, ``L`` is the specific latent heat of the phase transition, which is ``L_v`` for the saturation vapor pressure over liquid, or ``L_s`` for the saturation vapor pressure over ice. Substituting the linear relation \eqref{e:LHTemperature} between latent heat and temperature, and taking ``p_\mathrm{tr}`` to be the vapor pressure at the triple point (by definition equal to the saturation vapor pressures both over liquid and ice), the Clausius-Clapeyron relation can be integrated to give a closed-form expression for the vapor pressure that is consistent with our thermodynamic assumptions:
```math
\begin{equation}
    p_v^* = p_{\mathrm{tr}} \left( \frac{T}{T_{\mathrm{tr}}} \right)^{\frac{\Delta c_p}{R_v}}
        \exp \left[ \frac{L_0 - \Delta c_p T_0}{R_v}
        \left( \frac{1}{T_{\mathrm{tr}}} - \frac{1}{T} \right) \right].
        \label{e:SatVaporPressure}
\end{equation}
```

With ``L_0 = L_{v,0}`` or ``L_0 = L_{s,0}`` and the corresponding heat capacity difference ``\Delta c_p``, this gives saturation vapor pressures over liquid or ice that are accurate within 3% for temperatures between 200K and 330K (with accuracy better than 1% for typical near-surface conditions).

> To obtain the saturation vapor pressure over a mixture of liquid and ice (e.g., in mixed-phase clouds), using a weighted average of the relevant specific latent heats in the vapor pressure \eqref{e:SatVaporPressure} leads to a thermodynamically consistent formulation [Pressel2015](@cite). That is, if a fraction ``λ_p`` of the condensate is liquid and the complement ``1-λ_p`` is ice, calculating the saturation vapor pressure with a specific latent heat ``λ_p L_v + (1-λ_p)L_s`` gives a thermodynamically consistent saturation vapor pressure over the mixture.

Reviewer: ```suggestion
To obtain the saturation vapor pressure over a mixture of liquid and ice (e.g., in mixed-phase clouds), using a weighted average of the relevant specific latent heats in the vapor pressure \eqref{e:SatVaporPressure} leads to a thermodynamically consistent formulation ([Pressel2015](@cite)). That is, if a fraction ``λ_p`` of the condensate is liquid and the complement ``1-λ_p`` is ice, calculating the saturation vapor pressure with a specific latent heat ``λ_p L_v + (1-λ_p)L_s`` gives a thermodynamically consistent saturation vapor pressure over the mixture.
```

TODO: there was a commented equation here, do we need it?

This expression for temperature as a function of liquid-ice potential temperature is obtained from \eqref{e:TempFromThetaLiGivenP} by substituting for pressure in the Exner function ``\Pi`` from the ideal gas law, ``p=ρ R_m T``, solving for temperature using a second-order Taylor expansion around ``T_u`` for small condensate specific humidities, and using the relation ``1-κ = c_{vm}/c_{pm}``, which follows from ``c_{pm} - R_m = c_{vm}``. The relation for temperature \eqref{e:TempFromThetaLiGivenRho} holds to second order in condensate specific humidities ``q_l`` and ``q_i``. That is, the inversion relation \eqref{e:TempFromThetaLiGivenRho} holds to one higher order of accuracy than the definition of the liquid-ice potential temperature \eqref{e:liquid_ice_pottemp} itself, which is only first-order accurate in the condensate specific humidities ``q_l`` and ``q_i``.

### Speed of Sound
The speed of sound in (moist) unstratified air is
```math
\begin{equation}
 c_s = \left(\frac{c_{pm}}{c_{vm}} R_m T \right)^{1/2},
\label{e:soundspeed}
\end{equation}
```

> with the appropriate gas constants for moist air. In the presence of stratification, additional terms arise [Durran1999](@cite).

Reviewer: ```suggestion
with the appropriate gas constants for moist air. In the presence of stratification, additional terms arise ([Durran1999](@cite)).
```
